cargo-near/src/abi.rs

```diff
@@ -53,15 +55,34 @@ pub(crate) fn write_to_file(
     if !generate_docs {
         strip_docs(&mut contract_abi);
     }
-    let near_abi_json = serde_json::to_string(&contract_abi)?;
-    let out_path_abi = crate_metadata
-        .target_directory
-        .join(crate_metadata.root_package.name.replace('-', "_") + "_abi.json");
-    fs::write(&out_path_abi, near_abi_json)?;
+    let near_abi_serialized = match format {
+        AbiFormat::Json => serde_json::to_vec_pretty(&contract_abi)?,
+        AbiFormat::JsonMin => serde_json::to_vec(&contract_abi)?,
+    };
+    let near_abi_compressed = match compression {
+        AbiCompression::NoOp => near_abi_serialized,
+        AbiCompression::Zstd => zstd::encode_all(near_abi_serialized.as_slice(), 0)?,
+    };
+
+    let out_path_abi = crate_metadata.target_directory.join(format!(
+        "{}_abi.{}",
```

Comment:
When building a contract, and embedding abi, this exports a compressed ABI file. Shouldn't ABI compression should be limited to contract embedding only, or wdyt? So the exported ABI file should be the same as if you ran `cargo near abi`, which is a pretty-printed json file.

Reply:
Just saw this was mentioned in your comment.

Reply:
See my comment above. It is not clear to me what the best flow would be here, seems opinionated.

Reply:
Honestly, makes me inclined not to expose `--format`/`--compression` for now and postpone this decision until we understand the use cases a little better.

Reply:
Opinionated, but I think.

1. We assume JSON as the ABI format. So there's no need for `--format`, seems redundant. Instead, we add `--pretty`.
2. We assume, when embedding, we always want to optimize for size. So compression is enabled by default. But we leave `--uncompress` to opt-out.

Reply:
> --pretty

Did you mean `--no-pretty` (or `--ugly` :grimacing:) or do you propose to minify json by default?

> So compression is enabled by default. But we leave --uncompress to opt-out.

Does it even makes sense to opt-out if all our tooling is going to expect this to be compressed?

Reply:
Let's just leave both out. And only have a format toggle on the abi command.

Reply:
Agreed, got rid of it